### PR TITLE
Feature/Localdev-update: Implement NtvSectionAdapter on GridView, ListView

### PR DESCRIPTION
**Pull request by K-thao**
- Converted some classes to **Kotlin**
- Implement `NtvSectionAdapter` on `GridViewAdapter`, `ListViewAdapter`
- Use common constant variable for **tag**.

### CODE REVIEW

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/ListViewAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        initialNativoRequest = false
+    }
+
+    override fun didAssignAdToLocation(
+        location: Int,
+        adData: NtvAdData,
+        inSection: String,
+        container: ViewGroup
+    ) {
+        Log.d(NtvTAG, "didAssignAdToLocation: $location")
```
Comment (MattMur) [inline app/src/main/java/com/nativo/sampleapp/ViewAdapter/ListViewAdapter.kt:111]: When an ad gets assigned to a location, you should add an extra row to your datasource(integerList) and then inject that row here

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        initialNativoRequest = false
+    }
+
+    override fun didAssignAdToLocation(
+        location: Int,
+        adData: NtvAdData,
+        inSection: String,
+        container: ViewGroup
+    ) {
+        Log.d(NtvTAG, "didAssignAdToLocation: $location")
```
Comment (MattMur) [inline app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt:117]: When an ad gets assigned to a location, you should add an extra row to your datasource(integerList) and then inject that row here

**Review by MattMur — CHANGES_REQUESTED** [review]
Needs a bit of work but looking good

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    companion object {
+        const val ITEM_COUNT = 30
+    }
+
+    var integerList: MutableList<Int> = ArrayList()
+    var initialNativoRequest = true
+
+    init {
+        // Nativo init
+        initSectionWithAdapter(this, SECTION_URL, context)
```
[the remark this answers is not in the document]
Comment (K-thao) [reply app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt]: Fixed.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+import android.widget.ArrayAdapter
+import android.widget.GridView
+import android.widget.ImageView
+import android.widget.TextView
+import com.nativo.sampleapp.R
+import com.nativo.sampleapp.util.AppConstants.CLICK_OUT_URL
+import com.nativo.sampleapp.util.AppConstants.NtvTAG
+import com.nativo.sampleapp.util.AppConstants.SECTION_URL
+import net.nativo.sdk.NativoSDK.initSectionWithAdapter
+import net.nativo.sdk.NativoSDK.placeAdInView
```
[the remark this answers is not in the document]
Comment (K-thao) [reply app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt]: Fixed.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        } else {
+            convertView = view
+            holder = view.tag as ItemViewHolder
+        }
+
+        holder.bind(integerList[i], i)
+        return convertView
+    }
+
+    inner class ItemViewHolder(private val container: View) {
```
[the remark this answers is not in the document]
Comment (K-thao) [reply app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt]: Fixed.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                    Intent(
+                        Intent.ACTION_VIEW,
+                        Uri.parse(CLICK_OUT_URL)
+                    )
+                )
+            }
+
+            if (shouldShowPlaceAd(item)) {
+                val isAdContentAvailable =
+                    placeAdInView(container, gridView, SECTION_URL, position, null)
```
[the remark this answers is not in the document]
Comment (K-thao) [reply app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt]: Fixed.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/ListViewAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
+        val convertView: View = view
+            ?: LayoutInflater.from(viewGroup.context)
+                .inflate(R.layout.publisher_article, viewGroup, false)
+        bindView(convertView, integerList[i], i)
+        return convertView
+    }
+
+    private fun bindView(container: View, item: Int, i: Int) {
```
[the remark this answers is not in the document]
Comment (K-thao) [reply app/src/main/java/com/nativo/sampleapp/ViewAdapter/ListViewAdapter.kt]: Fixed.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 8]
```diff
@@ -96,6 +103,10 @@ class GridViewAdapter(context: Context, private val gridView: GridView) :
         }
     }
 
+    fun setItemClickListener(listener: ((item: Int) -> Unit)?) {
+        _itemClickListener = listener
+    }
+
```
Comment (MattMur) [inline app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt:109]: What is this for? I don't see this used anywhere.

File: app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt [hunk 9]
```diff
@@ -76,12 +88,7 @@ class GridViewAdapter(context: Context, private val gridView: GridView) :
...
-                context.startActivity(
-                    Intent(
-                        Intent.ACTION_VIEW,
-                        Uri.parse(CLICK_OUT_URL)
-                    )
-                )
+                _itemClickListener?.invoke(item)
             }
 
             if (shouldShowPlaceAd(item)) {
```
Comment (MattMur) [inline app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt:94]: What I would recommend is have an 'else' condition after this, so if 'shouldShowPlaceAd()' is false, then you add a click listener.

**Review by MattMur — CHANGES_REQUESTED** [review]

**Review by MattMur — APPROVED** [review]

**Pull request merged**